To: vim_dev@googlegroups.com
Subject: Patch 7.3.291
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.291
Problem:    Configure doesn't work properly with Python3.
Solution:   Put -ldl before $LDFLAGS. Add PY3_NO_RTLD_GLOBAL. (Roland
	    Puntaier)
Files:	    src/config.h.in, src/auto/configure, src/configure.in


*** ../vim-7.3.290/src/config.h.in	2010-10-27 16:49:41.000000000 +0200
--- src/config.h.in	2011-09-02 11:22:21.000000000 +0200
***************
*** 346,351 ****
--- 346,354 ----
  /* Define if dynamic python does not require RTLD_GLOBAL */
  #undef PY_NO_RTLD_GLOBAL
  
+ /* Define if dynamic python3 does not require RTLD_GLOBAL */
+ #undef PY3_NO_RTLD_GLOBAL
+ 
  /* Define if you want to include the Ruby interpreter. */
  #undef FEAT_RUBY
  
*** ../vim-7.3.290/src/auto/configure	2011-07-15 13:09:46.000000000 +0200
--- src/auto/configure	2011-09-02 11:23:15.000000000 +0200
***************
*** 5733,5739 ****
    cflags_save=$CFLAGS
    CFLAGS="$CFLAGS $PYTHON_CFLAGS"
    ldflags_save=$LDFLAGS
!   LDFLAGS="$LDFLAGS -ldl"
    if test "$cross_compiling" = yes; then :
    { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
  $as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
--- 5733,5739 ----
    cflags_save=$CFLAGS
    CFLAGS="$CFLAGS $PYTHON_CFLAGS"
    ldflags_save=$LDFLAGS
!     LDFLAGS="-ldl $LDFLAGS"
    if test "$cross_compiling" = yes; then :
    { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
  $as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
***************
*** 5798,5804 ****
    cflags_save=$CFLAGS
    CFLAGS="$CFLAGS $PYTHON3_CFLAGS"
    ldflags_save=$LDFLAGS
!   LDFLAGS="$LDFLAGS -ldl"
    if test "$cross_compiling" = yes; then :
    { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
  $as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
--- 5798,5804 ----
    cflags_save=$CFLAGS
    CFLAGS="$CFLAGS $PYTHON3_CFLAGS"
    ldflags_save=$LDFLAGS
!     LDFLAGS="-ldl $LDFLAGS"
    if test "$cross_compiling" = yes; then :
    { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
  $as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
*** ../vim-7.3.290/src/configure.in	2011-07-15 13:09:46.000000000 +0200
--- src/configure.in	2011-09-02 11:19:51.000000000 +0200
***************
*** 1161,1167 ****
    cflags_save=$CFLAGS
    CFLAGS="$CFLAGS $PYTHON_CFLAGS"
    ldflags_save=$LDFLAGS
!   LDFLAGS="$LDFLAGS -ldl"
    AC_RUN_IFELSE([
      #include <dlfcn.h>
      /* If this program fails, then RTLD_GLOBAL is needed.
--- 1161,1168 ----
    cflags_save=$CFLAGS
    CFLAGS="$CFLAGS $PYTHON_CFLAGS"
    ldflags_save=$LDFLAGS
!   dnl -ldl must go first to make this work on Archlinux (Roland Puntaier)
!   LDFLAGS="-ldl $LDFLAGS"
    AC_RUN_IFELSE([
      #include <dlfcn.h>
      /* If this program fails, then RTLD_GLOBAL is needed.
***************
*** 1205,1211 ****
    cflags_save=$CFLAGS
    CFLAGS="$CFLAGS $PYTHON3_CFLAGS"
    ldflags_save=$LDFLAGS
!   LDFLAGS="$LDFLAGS -ldl"
    AC_RUN_IFELSE([
      #include <dlfcn.h>
      #include <wchar.h>
--- 1206,1213 ----
    cflags_save=$CFLAGS
    CFLAGS="$CFLAGS $PYTHON3_CFLAGS"
    ldflags_save=$LDFLAGS
!   dnl -ldl must go first to make this work on Archlinux (Roland Puntaier)
!   LDFLAGS="-ldl $LDFLAGS"
    AC_RUN_IFELSE([
      #include <dlfcn.h>
      #include <wchar.h>
*** ../vim-7.3.290/src/version.c	2011-09-02 11:56:15.000000000 +0200
--- src/version.c	2011-09-02 12:25:13.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     291,
  /**/

-- 
DENNIS: Look,  strange women lying on their backs in ponds handing out
        swords ... that's no basis for a system of government.  Supreme
        executive power derives from a mandate from the masses, not from some
        farcical aquatic ceremony.
                 "Monty Python and the Holy Grail" PYTHON (MONTY) PICTURES LTD

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
